dashboards-observability/public/components/visualizations/charts/box_plot/box_plot.tsx

```diff
@@ -0,0 +1,190 @@
+
+const { PointPosition } = DefaultBoxChartStyles;
+export const BoxPlot = ({ visualizations, layout, config }: any) => {
+  const DEFAULT_LABEL_SIZE = 10;
+  const { vis } = visualizations;
+  const {
+    data,
+    metadata: { fields },
+  } = visualizations.data.rawVizData;
+  const lastIndex = fields.length - 1;
+  const {
+    dataConfig = {},
+    layoutConfig = {},
+    availabilityConfig = {},
+  } = visualizations?.data?.userConfigs;
+  const valueForXSeries = dataConfig?.valueOptions?.dimensions
+    ? filterDataConfigParameter(dataConfig.valueOptions.dimensions)
+    : [];
+  const valueSeries = dataConfig?.valueOptions?.metrics
+    ? filterDataConfigParameter(dataConfig.valueOptions.metrics)
+    : [];
+  const boxOrientation = dataConfig?.chartStyles?.orientation || vis.orientation;
+  const isVertical = boxOrientation === vis.orientation;
+  const boxMode = dataConfig?.chartStyles?.boxMode || vis.boxmode;
+  let box;
+
+  if (isEmpty(valueSeries) || (boxMode === 'group' && valueForXSeries.length === 0)) {
+    return <EmptyPlaceholder icon={visualizations?.vis?.icontype} />;
+  }
+
+  const tickAngle = dataConfig?.chartStyles?.rotateBoxLabels || vis.labelangle;
+  const markerSize = dataConfig?.chartStyles?.markerSize || vis.markersize;
+  const fillOpacity =
+    dataConfig?.chartStyles?.fillOpacity !== undefined
+      ? dataConfig?.chartStyles?.fillOpacity / FILLOPACITY_DIV_FACTOR
+      : vis.fillopacity / FILLOPACITY_DIV_FACTOR;
+  const showLegend = !(
+    dataConfig?.legend?.showLegend && dataConfig.legend.showLegend !== vis.showlegend
+  );
+  const legendPosition = dataConfig?.legend?.position || vis.legendposition;
+  const labelSize = dataConfig?.chartStyles?.labelSize || DEFAULT_LABEL_SIZE;
+  const boxGap = dataConfig?.chartStyles?.boxGap || vis.boxgap;
+  const jitter = dataConfig?.chartStyles?.jitter || vis.jitter;
+  const legendSize = dataConfig?.legend?.legendSize;
+
+  const getSelectedColorTheme = (field: ConfigListEntry, index: number) =>
+    (dataConfig?.colorTheme?.length > 0 &&
+      dataConfig.colorTheme.find((colorSelected: any) => colorSelected?.name.name === field.label)
+        ?.color) ||
+    PLOTLY_COLOR[index % PLOTLY_COLOR.length];
+
+  const dimensionData = valueForXSeries.reduce((prev, cur) => {
+    if (prev.length === 0) {
+      return data[cur.name].flat();
+    }
+    return prev.map(
+      (item: string | number, index: number) => `${item},<br>${data[cur.name][index]}`
+    );
+  }, []);
+
+  box = valueSeries.map((field: ConfigListEntry, index: number) => {
+    const selectedColor = getSelectedColorTheme(field, index);
+    const axis = isVertical ? 'y' : 'x';
+    const alternateAxis = isVertical ? 'x' : 'y';
+    return {
+      [axis]: data[field.name],
+      ...(boxMode === 'group' && {
+        [alternateAxis]: dimensionData,
+        orientation: boxOrientation,
+      }),
+      boxpoints: 'all',
+      jitter: jitter,
```

Comment:
what is significance of this property.

Reply:
jitter is a kind of random number to avoid overlapping of dots. 
in boxpoints key we are providing type of points we want to plot in box. we have few options here like: all, outliers, suspectedoutliers, false. each with different meaning @abasatwar